helper/chunkify_fix.js

```diff
@@ -0,0 +1,40 @@
+function createChunks(message, maxChars){
+    if (maxChars < 1){
+        return null;
+    }
+    let chunks = [];
+
+    if (message.length <= maxChars){
+        chunks.push(message);
+        return chunks;
+    }
+
+    const firstSection = message.slice(0, maxChars);
+
+    const nextlineIdx = firstSection.lastIndexOf('\n')
+    const carriageIdx = firstSection.lastIndexOf('\r')
+    const whitespaceIdx = firstSection.lastIndexOf(' ');
+    const maxIdx = Math.max(nextlineIdx, carriageIdx); //computes the max between the last occurrence of carriage return and newline
+
+    let currLine = '', remainder = '';
+    
+    if (maxIdx > 0){
+        currLine = message.slice(0, maxIdx);

```

Comment:
duplicated code in these 3 cases. Better to have an int splitIdx that is set here, then do the slicing once outside